### PR TITLE
feat(hpack): improve headers storage API

### PR DESCRIPTION
**Pull request by 3Hren**
@antmat TAL

### CODE REVIEW

File: include/cocaine/hpack/header.hpp [hunk 1]
```diff
@@ -191,14 +191,18 @@ class header_t {
 };
 
 class header_storage_t {
+    header_t::zone_t zone;
```
Comment (antmat) [inline include/cocaine/hpack/header.hpp]: Personally I don't like this style, as in header public api of the class is more important than it's layout.

File: include/cocaine/hpack/header.hpp [hunk 2]
```diff
@@ -191,14 +191,18 @@ class header_t {
...
+    std::vector<header_t> headers;
+
 public:
+    explicit header_storage_t() = default;
```
Comment (antmat) [inline include/cocaine/hpack/header.hpp]: why it is explicit?

File: include/cocaine/hpack/header.hpp (hunk 1 above)
Comment (3Hren) [inline include/cocaine/hpack/header.hpp]: There are no personality: https://github.com/cocaine/cocaine-core/wiki/codestyle

File: include/cocaine/hpack/header.hpp (hunk 2 above)
Comment (3Hren) [inline include/cocaine/hpack/header.hpp]: Oops

**Comment by 3Hren** [comment]
Updated.

**Comment by 3Hren** [comment]
Updated again

**Comment by antmat** [comment]
LGTM

**Pull request merged**